Marlin/src/gcode/bedlevel/abl/G29.cpp

```diff
@@ -683,7 +687,18 @@ void GcodeSuite::G29() {
           #endif
 
           measured_z = faux ? 0.001 * random(-100, 101) : probe_pt(xProbe, yProbe, raise_after, verbose_level);
-
+          
+          #if ENABLED(G29_RETRY_AND_RECOVER)
+            while (isnan(measured_z) && retries--) { //retry probe
+              #ifdef G29_ACTION_ON_RECOVER
+                SERIAL_ECHOLNPGM("//action:" G29_ACTION_ON_RECOVER);
+              #endif
+              #ifdef G29_RECOVER_COMMANDS
+                process_subcommands_now_P(PSTR(G29_RECOVER_COMMANDS));
+              #endif
+              measured_z = probe_pt(xProbe, yProbe, raise_after, verbose_level);
```

Comment:
Here we must be careful. If the probe failed, driving the nozzle into the bed, this might drive it even further into the bed. I suppose this might also be possible with the previous implementation too….

Reply:
I don't think this is a huge concern. As called by G29, `probe_pt` currently raises by `Z_CLEARANCE_BETWEEN_PROBES` whether or not the probe was successful. The nozzle would have to have been driven down by more than this for it to remain touching the bed. If this is a concern, the best way to fix this for all use cases would be to have probe_pt completely undo any failed probes before returning.